products/sle/main.pm

```diff
@@ -121,6 +121,13 @@ sub cleanup_needles {
     else {
         unregister_needle_tags('ENV-OFW-1');
     }
+
+    if (get_var('PXEBOOT')) {
+        unregister_needle_tags('ENV-PXEBOOT-0');
+    }
+    else {
+        unregister_needle_tags('ENV-PXEBOOT-1');
+    }
```

Comment:
ok, so let's keep it like this for now. You can try to simplify all the calls if you like as a refactoring for later.